File: src/main/java/com/chaottic/nbt/number/NumberTag.java
package com.chaottic.nbt.number;

import com.chaottic.nbt.Tag;

public sealed interface NumberTag extends Tag permits ByteTag, DoubleTag, FloatTag, IntTag, LongTag, ShortTag {

    byte getAsByte();

    short getAsShort();

    int getAsInt();

    long getAsLong();

    float getAsFloat();

    double getAsDouble();
}

File: src/main/kotlin/com/chaottic/nbt/number/NumberTag.kt
package com.chaottic.nbt.number

import com.chaottic.nbt.Tag

sealed interface NumberTag : Tag {
	fun asByte(): Byte
	fun asShort(): Short
	fun asInt(): Int
	fun asLong(): Long
	fun asFloat(): Float
	fun asDouble(): Double
}